src/main/java/com/agricraft/agricore/registry/AgriSoils.java

```diff
@@ -51,8 +53,8 @@ public boolean acceptsElement(String filename) {
     }
 
     @Override
-    public Class<AgriSoil> getElementClass() {
-        return AgriSoil.class;
+    public AgriJsonVersion<AgriSoil> getElementVersion() {
+        return Versions_1_16.SOIL;
```

Comment:
I like this.

Reply:
I'm sure something will pop up sooner or later that will break this...